x-pack/plugins/timelines/public/components/t_grid/body/index.tsx

```diff
@@ -633,39 +633,48 @@ export const BodyComponent = React.memo<StatefulBodyProps>(
       return Cell;
     }, [columnHeaders, data, id, renderCellValue, tabType, theme, browserFields, rowRenderers]);
 
+    // Store context in state rather than creating object in provider value={} to prevent re-renders caused by a new object being created
+    const [activeStatefulEventContext] = useState({
+      timelineID: id,
+      tabType,
+      enableHostDetailsFlyout: true,
+      enableIpDetailsFlyout: true,
+    });
     return (
       <>
-        {tableView === 'gridView' && (
-          <EuiDataGrid
-            data-test-subj="body-data-grid"
-            aria-label={i18n.TGRID_BODY_ARIA_LABEL}
-            columns={columnsWithCellActions}
-            columnVisibility={{ visibleColumns, setVisibleColumns }}
-            gridStyle={gridStyle}
-            leadingControlColumns={leadingTGridControlColumns}
-            trailingControlColumns={trailingTGridControlColumns}
-            toolbarVisibility={toolbarVisibility}
-            rowCount={data.length}
-            renderCellValue={renderTGridCellValue}
-            inMemory={{ level: 'sorting' }}
-            sorting={{ columns: sortingColumns, onSort }}
-          />
-        )}
-        {tableView === 'eventRenderedView' && (
-          <EventRenderedView
-            alertToolbar={alertToolbar}
-            browserFields={browserFields}
-            events={data}
-            leadingControlColumns={leadingTGridControlColumns ?? []}
-            onChangePage={loadPage}
-            pageIndex={activePage}
-            pageSize={querySize}
-            pageSizeOptions={itemsPerPageOptions}
-            rowRenderers={rowRenderers}
-            timelineId={id}
-            totalItemCount={totalItems}
-          />
-        )}
+        <StatefulEventContext.Provider value={activeStatefulEventContext}>
```

Comment:
I found that we didn’t provide the necessary info to t-grid, so it doesn’t know it should expand host and ip details flyout, so I added it in here.

Reply:
But it also means that I moved StatefulEventContext to timeline's plugin and import it from there, not sure if it's the right thing to do.